docker-app/qfieldcloud/core/views/files_views.py

```diff
@@ -135,25 +139,54 @@ def get(self, request, projectid, filename):
 
     # TODO refactor this function by moving the actual upload and Project model updates to library function outside the view
     def post(self, request, projectid, filename, format=None):
-        project = Project.objects.get(id=projectid)
+        """Handle POST requests."""
+
+        # FIXME: CURRENTLY UNDERGOING DEBUG
+        # Write traceback to buffer + deepcopy request before anything happens to the request object
+        buffer = io.StringIO()
+        print_stack(limit=50, file=buffer)
+
+        # Fetch important data without mutating request
+        request_attributes = {
+            "data": str(copy.copy(request.data).keys()),
+            "files": str(request.FILES.keys()),
+            "meta": str(request.META),
+        }
+
+        if len(request.FILES.getlist("file")) > 1:
+            raise exceptions.MultipleContentsError()
```

Comment:
The tests suggest it's not breaking things. The goal is to make explicit an assumption of the API, namely, that exactly one file should be uploaded from this view. I hope I didn't make this assumption up...

Reply:
If there's the slightest doubt, I'd prefer to report something like this to sentry and let it go through, but I remember @suricactus mentioning 100% ... So ...

Reply:
Theoretically it is possible to upload multiple files, even though it should never happen. While it would be safer to just log it, I think we are pretty sure enforcing it with an exception is good.

Anyways, this is not directly related to the issue we are trying to solve, because we should have seen it in the already existing logs.